modules/host/contractmanager/storagefolderadd.go

```diff
@@ -6,9 +6,42 @@ import (
 	"sync/atomic"
 
 	"github.com/NebulousLabs/Sia/build"
+	"github.com/NebulousLabs/Sia/crypto"
 	"github.com/NebulousLabs/Sia/modules"
 )
 
+// findUnfinishedStorageFolderAdditions will scroll through a set of state
+// changes and figure out which of the unfinished storage folder additions are
+// still unfinished. If a storage folder addition has finished, it will be
+// discoverable through the presence of a storage folder addition object using
+// the same path, or through an errored storage folder addition object using
+// the same path. Putting the logic here keeps the WAL logic cleaner.
+func findUnfinishedStorageFolderAdditions(scs []stateChange) []*storageFolder {
+	// Create a map containing a list of all unfinished storage folder
+	// additions identified by their path to the storage folder that is being
+	// added.
+	usfMap := make(map[uint16]*storageFolder)
+	for _, sc := range scs {
+		for _, sf := range sc.UnfinishedStorageFolderAdditions {
+			usfMap[sf.Index] = sf
+		}
+		for _, sf := range sc.StorageFolderAdditions {
+			delete(usfMap, sf.Index)
+		}
```

Comment:
how does the ordering work here? Is looks like if you have these state changes:
```
scs[0] = {
    Unfinished: [1, 2, 3]
    Additions: [3, 4]
}
scs[1] = {
    Unfinished: [4, 5]
    Additions: [1]
}
```
Then `scs[1]` will delete addition 1, but `scs[0]` will not delete addition 4 (making the final state `[2, 5]`). That behavior seems reasonable to me, but I wanted to double-check. Is it not possible for `scs[0]` to reference an unfinished addition in `scs[1]`?